include/eve/arch/cpu/top_bits.hpp

```diff
@@ -106,32 +106,32 @@ namespace detail
       else                            return movemask(logical_type{}).first;
     }
 
-    EVE_FORCEINLINE static constexpr std::ptrdiff_t bits_per_element_impl()
+    EVE_FORCEINLINE static consteval std::ptrdiff_t bits_per_element_impl()
     {
       if constexpr ( is_aggregated ) return top_bits<half_logical>::bits_per_element;
       else                           return decltype(movemask(logical_type{}).second){}();
     }
 
-    static constexpr bool is_cheap_impl()
+    static consteval bool is_cheap_impl()
```

Comment:
same story here - we don't want the ifs to be constexpr when t doesn't have to. And I don't like consteval for no reason.